awx/main/models/jobs.py

```diff
@@ -1088,6 +1088,53 @@ def save(self, *args, **kwargs):
         super(JobHostSummary, self).save(*args, **kwargs)
 
 
+class UniqueHost(CreatedModifiedModel):
+    """
+    Retention of unique host targets (optional)
+    """
+
+    class Meta:
+        app_label = 'main'
+        constraints = [models.UniqueConstraint(name='ensure_unique_hosts', fields=['host_name'])]
```

Comment:
Is this preferredd rather than setting it on the `host` field below as `unique=True` ?

```
host = models.ForeignKey(
  'Host',
  [...]
  unique=True,
  [...]
```

Reply:
The reason for the new table to begin with is that the hosts and jobs can be deleted anytime, at which point they are removed entirely from the DB.  This table does have links to the host and job summary tables, but just for information.  When those hosts/jobs are deleted, the links are removed but the rows remain in this table until specifically cleaned up.  So yes that would work, but it wouldn't meet the goal.  The unique constraint is placed on the hostname field since it is independent of the foreign keys..

Reply:
Honesty, I was asking more about the *semantics* of defining the unique constraint there rather than at the table definition

Reply:
Yeah that is fine, but the important data is the host_name field, which is not a foreign key (unlike the host field which is a pointer to the host table).  Based on the subsequent comments and risk of impacting performance, I'm thinking maybe just removing the links to the host and job summary tables entirely is the way to go.  In that case it will just be host_name and recorded date, which don't have any dependency on other tables.